bitcoin/src/blockdata/locktime/relative.rs

```diff
@@ -47,6 +47,50 @@ impl LockTime {
     /// The number of bytes that the locktime contributes to the size of a transaction.
     pub const SIZE: usize = 4; // Serialized length of a u32.
 
+    /// Constructs a `LockTime` from an nSequence value or the argument to OP_CHECKSEQUENCEVERIFY.
+    ///
+    /// This method will **not** round-trip with [`Self::to_consensus_u32`], because relative
+    /// locktimes only use some bits of the underlying `u32` value and discard the rest. If
+    /// you want to preserve the full value, you should use the [`Sequence`] type instead.
```

Comment:
Yeah, I considered this. I could go either way on it. I'm not sure that making users go through `Sequence` makes this flow any less footgunny.

But I can remove this if you feel strongly about it.

I'd maybe like to add a method to script to encode a `Sequence` number directly into script, if there isn't one, so that users would never need a `u32`.